src/app/(pages)/authors/[slug]/page.tsx

```diff
@@ -0,0 +1,50 @@
+"use client";
+
+import React from "react";
+
+import { notFound } from "next/navigation";
+import fetchContentFromAuthor from "@/app/_api/fetchContentFromAuthor";
+import { fetchDoc } from "@/app/_api/fetchDoc";
+
+
+export default async function ContributorPage({ params: { slug } }) {
+
+  let contributor = null;
+  let postsFromContributor = null
+
+  try {
+    contributor = await fetchDoc({
```

Comment:
A couple of things: this could be simplified by moving the error handling into fetchDoc. Like that the logic in the component could just be:
```
const contributor = await fetchDoc(...)
const postsFromContributor = await fetchContentFromAuthor(...)
```
As this will happen in each page, it would make things simpler. Also, the return value of fetchDoc isn't typed, it's just `unknown`, which isn't very useful.

But that's a bit out of the scope here, so I would recommend merging the current PRs first, and we can get back to this afterwards.

Reply:
Thanks. I'll move this to an issue.